### PR TITLE
[ISSUE-67015020] query by config id, type filter for "/configurations"

### PR DESCRIPTION
**Pull request by haomingzhang**

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -231,6 +238,47 @@ func (a *apiManager) apiReturnBlobData(w http.ResponseWriter, r *http.Request) {
 
 }
 
+func (a *apiManager) apiHandleConfigId(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	configId := vars["configId"]
+	if !isValidUuid(configId) {
+		a.writeError(w, http.StatusBadRequest, API_ERR_BAD_CONFIG_ID, "configuration id is invalid")
```
Comment (sramamoorthy73) [inline api.go]: Change the verbiage as `configuration id is not in valid uuid format`

File: api.go [hunk 2]
```diff
@@ -380,6 +433,11 @@ func convertTime(t string) string {
 	return t
 }
 
+func isValidUuid(uuid string) bool {
+	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
+	return r.MatchString(uuid)
+}
+
```
Comment (sramamoorthy73) [inline api.go]: INstead of every plugin defining it locally, can we move this to the core module?

File: api.go [hunk 3]
```diff
@@ -102,14 +107,15 @@ type apiManagerInterface interface {
...
-	eTag                int64
-	deploymentsChanged  chan interface{}
-	addSubscriber       chan chan deploymentsResult
-	removeSubscriber    chan chan deploymentsResult
-	apiInitialized      bool
+	dbMan                dbManagerInterface
+	deploymentsEndpoint  string
+	blobEndpoint         string
+	deploymentIdEndpoint string
+	eTag                 int64
```
Comment (rajanishgj) [inline api.go:113]: can we use different name than etag

File: api.go [hunk 4]
```diff
@@ -231,6 +238,47 @@ func (a *apiManager) apiReturnBlobData(w http.ResponseWriter, r *http.Request) {
 
 }
 
+func (a *apiManager) apiHandleConfigId(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	configId := vars["configId"]
+	if !isValidUuid(configId) {
+		a.writeError(w, http.StatusBadRequest, API_ERR_BAD_CONFIG_ID, "configuration id is invalid")
+		return
```
Comment (rajanishgj) [inline api.go]: This configId is generated by mgmt or any other component. Its not good idea to assume its going to be UUID all the time. I think we have to relax this validation

**Review by sramamoorthy73 — APPROVED** [review]

**Review by rajanishgj — COMMENTED** [review]
LGTM

**Pull request merged**